server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java

```diff
@@ -81,14 +83,25 @@ CassandraSessionConfiguration getCassandraSessionConfiguration(FileSystem fileSy
 
     @Provides
     @Singleton
-    Cluster provideCluster(CassandraSessionConfiguration cassandraSessionConfiguration, AsyncRetryExecutor executor) throws FileNotFoundException, ConfigurationException, ExecutionException, InterruptedException {
+    ClusterFactory.QueryLoggerConf getCassandraQueryLoggerConf() {
+        return ClusterFactory.QueryLoggerConf.builder()
+            .withConstantThreshold(500)
+            .build();
+    }
+
+    @Provides
+    @Singleton
+    Cluster provideCluster(CassandraSessionConfiguration cassandraSessionConfiguration, AsyncRetryExecutor executor, ClusterFactory.QueryLoggerConf queryLoggerConf) throws FileNotFoundException, ConfigurationException, ExecutionException, InterruptedException {
```

Comment:
Hmmm Optionals are a little tricky but doable in Guice.

You can use @Name annotation, and then declare injection like : 

    bind(..).annotatedWith(Names.named("myName").to(...);

You might also need to handle that as generics (NameLitteral if my memory is good).

However, what's the troubble with always setting a QueryLoggerConf ? (Can't we get a default one ?)